Generated unit tests fail with gapic-generator-python 1.18.3 and google-api-core 1.34.1

In [gapic-generator 1.18.3](https://pypi.org/project/gapic-generator/1.18.3/), unit tests fail with error ` 'google.api_core.retry' has no attribute 'AsyncRetry'` when using `google-api-core==1.34.1` 

```
==================================== ERRORS ====================================
___ ERROR collecting tests/unit/gapic/managedkafka_v1/test_managed_kafka.py ____
google/cloud/managedkafka_v1/services/managed_kafka/pagers.py:33: in <module>
    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
E   AttributeError: module 'google.api_core.retry' has no attribute 'AsyncRetry'

During handling of the above exception, another exception occurred:
tests/unit/gapic/managedkafka_v1/test_managed_kafka.py:60: in <module>
    from google.cloud.managedkafka_v1.services.managed_kafka import (
google/cloud/managedkafka_v1/__init__.py:21: in <module>
    from .services.managed_kafka import ManagedKafkaAsyncClient, ManagedKafkaClient
google/cloud/managedkafka_v1/services/managed_kafka/__init__.py:16: in <module>
    from .async_client import ManagedKafkaAsyncClient
google/cloud/managedkafka_v1/services/managed_kafka/async_client.py:54: in <module>
    from google.cloud.managedkafka_v1.services.managed_kafka import pagers
google/cloud/managedkafka_v1/services/managed_kafka/pagers.py:36: in <module>
    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
E   AttributeError: module 'google.api_core.retry' has no attribute 'AsyncRetry'
- generated xml file: /home/runner/work/google-cloud-python/google-cloud-python/packages/google-cloud-managedkafka/unit_3.7_sponge_log.xml -
=========================== short test summary info ============================
ERROR tests/unit/gapic/managedkafka_v1/test_managed_kafka.py - AttributeError: module 'google.api_core.retry' has no attribute 'AsyncRetry'
```

--- a/gapic/templates/%namespace/%name_%version/%sub/services/%service/pagers.py.j2
+++ b/gapic/templates/%namespace/%name_%version/%sub/services/%service/pagers.py.j2
@@ -9,13 +9,14 @@
  #}
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 {% filter sort_lines %}
 {% for method in service.methods.values() | selectattr('paged_result_field') %}

--- a/gapic/templates/noxfile.py.j2
+++ b/gapic/templates/noxfile.py.j2
@@ -183,8 +183,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

--- a/noxfile.py
+++ b/noxfile.py
@@ -273,8 +273,6 @@ def showcase_library(
             # Install the library without a constraints file.
             session.install("-e", tmp_dir)
 
-        # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-        session.install("google-api-core>=2.19.1rc0")
         yield tmp_dir
 
 

--- a/tests/integration/goldens/asset/google/cloud/asset_v1/services/asset_service/pagers.py
+++ b/tests/integration/goldens/asset/google/cloud/asset_v1/services/asset_service/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.cloud.asset_v1.types import asset_service
 from google.cloud.asset_v1.types import assets

--- a/tests/integration/goldens/asset/noxfile.py
+++ b/tests/integration/goldens/asset/noxfile.py
@@ -194,8 +194,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

--- a/tests/integration/goldens/credentials/noxfile.py
+++ b/tests/integration/goldens/credentials/noxfile.py
@@ -194,8 +194,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

--- a/tests/integration/goldens/eventarc/google/cloud/eventarc_v1/services/eventarc/pagers.py
+++ b/tests/integration/goldens/eventarc/google/cloud/eventarc_v1/services/eventarc/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.cloud.eventarc_v1.types import channel
 from google.cloud.eventarc_v1.types import channel_connection

--- a/tests/integration/goldens/eventarc/noxfile.py
+++ b/tests/integration/goldens/eventarc/noxfile.py
@@ -194,8 +194,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

--- a/tests/integration/goldens/logging/google/cloud/logging_v2/services/config_service_v2/pagers.py
+++ b/tests/integration/goldens/logging/google/cloud/logging_v2/services/config_service_v2/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.cloud.logging_v2.types import logging_config
 

--- a/tests/integration/goldens/logging/google/cloud/logging_v2/services/logging_service_v2/pagers.py
+++ b/tests/integration/goldens/logging/google/cloud/logging_v2/services/logging_service_v2/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.api import monitored_resource_pb2  # type: ignore
 from google.cloud.logging_v2.types import log_entry

--- a/tests/integration/goldens/logging/google/cloud/logging_v2/services/metrics_service_v2/pagers.py
+++ b/tests/integration/goldens/logging/google/cloud/logging_v2/services/metrics_service_v2/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.cloud.logging_v2.types import logging_metrics
 

--- a/tests/integration/goldens/logging/noxfile.py
+++ b/tests/integration/goldens/logging/noxfile.py
@@ -194,8 +194,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

--- a/tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/pagers.py
+++ b/tests/integration/goldens/redis/google/cloud/redis_v1/services/cloud_redis/pagers.py
@@ -15,13 +15,14 @@
 #
 from google.api_core import gapic_v1
 from google.api_core import retry as retries
+from google.api_core import retry_async as retries_async
 from typing import Any, AsyncIterator, Awaitable, Callable, Sequence, Tuple, Optional, Iterator, Union
 try:
     OptionalRetry = Union[retries.Retry, gapic_v1.method._MethodDefault, None]
-    OptionalAsyncRetry = Union[retries.AsyncRetry, gapic_v1.method._MethodDefault, None]
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, gapic_v1.method._MethodDefault, None]
 except AttributeError:  # pragma: NO COVER
     OptionalRetry = Union[retries.Retry, object, None]  # type: ignore
-    OptionalAsyncRetry = Union[retries.AsyncRetry, object, None]  # type: ignore
+    OptionalAsyncRetry = Union[retries_async.AsyncRetry, object, None]  # type: ignore
 
 from google.cloud.redis_v1.types import cloud_redis
 

--- a/tests/integration/goldens/redis/noxfile.py
+++ b/tests/integration/goldens/redis/noxfile.py
@@ -194,8 +194,6 @@ def mypy(session):
         'types-protobuf'
     )
     session.install('.')
-    # Remove once https://github.com/googleapis/python-api-core/pull/650 is merged
-    session.install("google-api-core>=2.19.1rc0")
     session.run(
         'mypy',
         '-p',

